### PR TITLE
Electron support!

### PR DESCRIPTION
**Pull request by ohnx**
Electron support seems stable enough now.

To run, use `./Electron.app/Contents/MacOS/ node-music-player/ electron`

I think.

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -10,57 +13,98 @@ var util = require(__dirname + '/util.js');
...
+
+// Electron Specific
+if(ELECTRON) {
+var app = require('app');  // Module to control application life.
+var BrowserWindow = require('browser-window');  // Module to create native browser window.
+var mainWindow = null;
+app.on('window-all-closed', function() {
+  // On OS X it is common for applications and their menu bar
+  // to stay active until the user quits explicitly with Cmd + Q
+  if (process.platform != 'darwin') {
```
Comment (benkaiser) [inline app.js]: What about other platforms?

File: app.js [hunk 2]
```diff
@@ -10,57 +13,98 @@ var util = require(__dirname + '/util.js');
 var mkdirp = require('mkdirp');
 var proxy = require('express-http-proxy');
 
-var app = express();
-app.http().io();
-app.io.set('authorization', function (handshakeData, accept) {
+var eapp;
```
Comment (benkaiser) [inline app.js]: Why rename everything to eapp? Why not rename all the electron stuff below to eapp so the rest of the file stays the same?

File: views/base/main_base.html [hunk 3]
```diff
@@ -61,6 +61,7 @@
       "/static/js/player/coverbox.js"
       );
   </script>
+  <script src="//code.jquery.com/jquery-1.11.3.min.js"></script>
```
Comment (benkaiser) [inline views/base/main_base.html]: work out a fix for this. I'm not going to merge with this change.

File: app.js (hunk 2 above)
Comment (ohnx) [inline app.js]: Ha! My thinking was eapp = express app.

I suppose eapp could also mean electron app...

File: app.js (hunk 1 above)
Comment (ohnx) [inline app.js]: On other platforms, this isn't needed.

**Comment by ohnx** [comment]
This shouldn't break anything that already exists, so it should be safe to pull.

File: app.js (hunk 2 above)
Comment (benkaiser) [inline app.js]: Yeah I got the thinking. I just feel we should try and keep things the same as much as possible (i.e. keeping the express app named `app` and then naming the electron app `eapp` or something).

File: views/base/main_base.html (hunk 3 above)
Comment (ohnx) [inline views/base/main_base.html]: Okay, I'll try and see what the main issue is.

**Comment by ohnx** [comment]
I just realized this was an accidental pull request. It isn't ready for public use yet (I meant to submit this to my own fork)

Oops.

**Comment by benkaiser** [comment]
Ohhhh okay all good.

Yeah I was about to have a rant about code quality :P

**Comment by ohnx** [comment]
commit names too :)

**Pull request closed without merging**